lib/client/api.go

```diff
@@ -2660,6 +2660,11 @@ func (tc *TeleportClient) applyProxySettings(proxySettings webclient.ProxySettin
 	}
 
 	tc.TLSRoutingEnabled = proxySettings.TLSRoutingEnabled
+	if tc.TLSRoutingEnabled {
+		// If proxy supports TLS Routing all k8s requests will be sent to the WebProxyAddr where TLS Routing will identify
+		// k8s requests by "kube." SNI prefix and route to the kube proxy service.
+		tc.KubeProxyAddr = tc.WebProxyAddr
```

Comment:
Can Kube public addr still be different than web proxy addr? I.e. they may route to the same TLS routing-enabled port on the proxy, but can somebody specify different `kube_public_addr`? Maybe use `KubeProxyHost` here too?

Reply:
I'm a bit confused,  based on the issue https://github.com/gravitational/teleport/issues/8952 regarding `tunnel_public_addr` and `web_listen_addr`  the WebProxyAddr should take precedence over TunnelPublicAddress is proxy is TLS Routing is enabled.  Does `KubeProxyAddr `behavior should be consistent with this approach ?

Reply:
@smallinsky Thinking more about this, I think you're right. Kube public addr is what older clients without TLS routing connect to, right? So yeah, it should be ok.

Reply:
Yes, that makes sense.  Also confirmed with @dboslee that expected behavior for the cloud is that `WebProxyAddr`(`web_public_proxy_adress`) takes precedence over `kube_public_address` and `tunnel_public_adress` is TLS routing is enabled in the proxy.